test/functional/terminal/buffer_spec.lua

```diff
@@ -410,6 +410,14 @@ describe('on_lines does not emit out-of-bounds line indexes when', function()
     feed_command('bdelete!')
     eq('', exec_lua([[return _G.cb_error]]))
   end)
+
+  it('runs TextChangedT event', function()
+    meths.set_var('called', 0)
+    command('autocmd TextChangedT * ++once let g:called = 1')
+    feed_command('terminal')
+    feed('iaa')
+    eq(1, meths.get_var('called'))
+  end)
```

Comment:
https://github.com/vim/vim/commit/4ccaedfcd7526983f4b6b3b06b0bfb54f333f1f3 has a test that checked behavior if the handler closed the buffer.